### ISSUE TITLE
Avoid generating implicit jumps in if statements when they unconditionally return/jump

### ISSUE THREAD

**Issue by Bananattack** [wiz-lang/wiz#115]
Currently, there is a small issue of suboptimal code generation when an `if` statement with an `else` clause contains a `return` statement in the `if` body. This appears to be either a regression, or an oversight, as this used to work in the legacy D versions of Wiz.

The `if` statement code generation should check if the last statement of the `if` statement body is an unconditional branch (goto, return, continue, break, etc. without a condition). If so, the if statement should omit any implicit branch at the end of the body:

In other words,

```
if carry {
    return;
} else {
  // do something
}
```

should become

```
    bcc @else
@if
    ret
@else
    ; do something
@end
```

rather than

```
    bcc @else
@if
    ret
    jmp @end
@else
    ; do something
@end
```

This will also fix the `6502_return_on_if_else_path.wiz` test included in the block tests.
